homeassistant/components/geo_json_events/geo_location.py

```diff
@@ -51,152 +51,123 @@
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the GeoJSON Events platform."""
+    coordinator = hass.data[DOMAIN][FEED][entry.entry_id]
+
+    @callback
+    def async_add_geolocation(coordinator, config_entry_unique_id, external_id):
+        """Add geolocation entity from feed."""
+        new_entity = GeoJsonLocationEvent(
+            coordinator, config_entry_unique_id, external_id
+        )
+        _LOGGER.debug("Adding geolocation %s", new_entity)
+        async_add_entities([new_entity], False)
+
+    coordinator.listeners.append(
+        async_dispatcher_connect(
+            hass, coordinator.async_event_new_entity(), async_add_geolocation
+        )
+    )
+    _LOGGER.debug("Geolocation setup done")
+
+
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the GeoJSON Events platform."""
-    url: str = config[CONF_URL]
-    scan_interval: timedelta = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
-    coordinates: tuple[float, float] = (
-        config.get(CONF_LATITUDE, hass.config.latitude),
-        config.get(CONF_LONGITUDE, hass.config.longitude),
-    )
-    radius_in_km: float = config[CONF_RADIUS]
-    # Initialize the entity manager.
-    manager = GeoJsonFeedEntityManager(
-        hass, async_add_entities, scan_interval, coordinates, url, radius_in_km
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
```

Comment:
Very quick scan, but a couple of things:
- I think it should raise an issue to tell the user to remove the manual configuration
- config_flow PRs are always quite big. I suggest that you remove everything else can can be kept for a follow-up PR (eg. the `sensor` platform)

Reply:
I followed your advice and reduced this change to the minimum, i.e. config flow only, and this is now available here: https://github.com/home-assistant/core/pull/87062